sdk/mx.sdk/mx_sdk_vm_impl.py

```diff
@@ -704,6 +704,7 @@ def _find_escaping_links(root_dir):
                     _add(layout, _component_base, 'dependency:{}/polyglot.config'.format(launcher_project), _component)
             for _library_config in sorted(_get_library_configs(_component), key=lambda c: c.destination):
                 graalvm_dists.update(_library_config.jar_distributions)
+                self.jimage_ignore_jars.update(_library_config.jar_distributions)
```

Comment:
I'm not sure i understand `jimage_ignore_jars` that seems a bit dangerous: what distribution is both on the jvmci/boot classpath and part of the jars of a launcher/library?

Reply:
This is needed so that we do not end up with a jlinked jdk that contains dummy module definitions of `org.graalvm.nativeimage.agent.jvmtibase`, `org.graalvm.nativeimage.agent.diagnostics`, `org.graalvm.nativeimage.agent.tracing`. Having them would break
```java
if (getBuilderModuleNames().contains(moduleName)) {
    /* Modules provided by the image-builder are not allowed on the -imagemp */
    String modulePathEntryStr = "module " + moduleName + " from " + modulePathEntry;
    showWarning("Ignoring " + modulePathEntryStr + " (implicitly provided by builder)");
    return;
}
``` in `com.oracle.svm.driver.NativeImage#addImageModulePath`

Reply:
> what distribution is both on the jvmci/boot classpath and part of the jars of a launcher/library?

Right, alternatively I could fix why we need this is the fist place.

Which is that currently the `native-image.properties` generated by `mx_sdk_vm_impl.NativePropertiesBuildTask.contents` have a `ImageClasspath/ImageModulePath` that contains all the transitive jars from the image builder itself. In case of using the module system this is not allowed and the jars from the image builder itself need to be filtered out (hence the filtering in `com.oracle.svm.driver.NativeImage#addImageModulePath`).

Reply:
OK, thanks for the explanation. I'll let you decide what's the best approach there.